src/connector.js

```diff
@@ -411,14 +417,21 @@ function initEngine() {
 			search: {
 				locale: params.lang,
 				searchHub: params.searchHub,
+				pipeline: params.pipeline
 			},
 			preprocessRequest: ( request, clientOrigin ) => {
 				try {
-					if ( clientOrigin === 'analyticsFetch' ) {
+					if( clientOrigin === 'analyticsFetch' || clientOrigin === 'analyticsBeacon'  ) {
 						let requestContent = JSON.parse( request.body );
 
 						// filter user sensitive content
 						requestContent.originLevel3 = params.originLevel3;
+
+						// documentAuthor cannot be longer than 128 chars
+						if ( requestContent.documentAuthor ){
+							requestContent.documentAuthor = requestContent.documentAuthor.substring( 0, 128 );
+						}
```

Comment:
Can you please document this change in the description of your PR? Or open a separate PR with this change, since it is unrelated to QS.